src/instance.rs

```diff
@@ -659,6 +659,19 @@ impl<'a, 'py, T> Borrowed<'a, 'py, T> {
         (*self).clone()
     }
 
+    /// Returns the raw FFI pointer represented by self.
+    ///
+    /// # Safety
+    ///
+    /// Callers are responsible for ensuring that the pointer does not outlive self.
+    ///
+    /// The reference is borrowed; callers should not decrease the reference count
+    /// when they are finished with the pointer.
+    #[inline]
+    pub fn as_ptr(self) -> *mut ffi::PyObject {
```

Comment:
A bit unrelated, but is there a particular reason why we didn't opt for `NonNull<PyObject>` for these pointer methods?

Reply:
I think it's a historical thing from before I was involved with PyO3. Given that mostly these pointers are just immediately passed to FFI, maybe it was more convenient? We could add non-null variants for cases where these pointers get handled by Rust I guess...